pkg/agent/agent.go

```diff
@@ -0,0 +1,159 @@
+// Package agent implements functions for injecting faults in a target
+package agent
+
+import (
+	"context"
+	"fmt"
+	"syscall"
+	"time"
+
+	"github.com/grafana/xk6-disruptor/pkg/agent/protocol"
+	"github.com/grafana/xk6-disruptor/pkg/agent/protocol/grpc"
+	"github.com/grafana/xk6-disruptor/pkg/agent/protocol/http"
+	"github.com/grafana/xk6-disruptor/pkg/runtime"
+)
+
+// Config maintains the configuration for the execution of the agent
+type Config struct {
+	Profiler *runtime.ProfilerConfig
+}
+
+// Agent maintains the state required for executing an agent command
+type Agent struct {
+	env    runtime.Environment
+	config *Config
+}
```

Comment:
I would split this object further. I think it makes a good holder for all the logic present in `do`, which I would keep almost exactly as it is: Create locks, handle signals, set up profiler, etc.

`ApplyHTTPDisruption` and `ApplyHTTPDisruption` could be different instances of a `Disruptor` interface, which `Agent` calls. This would make all objects less stateful and short lived (which I think it's always a good thing).

Reply:
> I would split this object further. 
I don't follow you here. Which object and what kind of split. Could you please elaborate?

Reply:
> ApplyHTTPDisruption and ApplyHTTPDisruption could be different instances of a Disruptor interface

I'm not sure I like this change. This makes the disruptor aware of the different protocols. The current design separates these concerns in two objects the generic protocol disruptor and the proxys for each protocol. I'm not in favor of changing this.